src/methods/geom_relations/geom_geom_processors.jl

```diff
@@ -0,0 +1,739 @@
+#= Code is based off of DE-9IM Standards (https://en.wikipedia.org/wiki/DE-9IM)
+and attempts a standardized solution for most of the functions.
+=#
+
+@enum PointOrientation point_in=1 point_on=2 point_out=3
+
+@enum LineOrientation line_cross=1 line_hinge=2 line_over=3 line_out=4
```

Comment:
Just for me to understand this, what is `line_hinge`?

Reply:
It is where the two lines form a hinge. For example, the below characters form a hinge.
|                             |
|--------       or     |
|                             |---------
It is basically where the endpoint of one line is somewhere on the other line (could be middle, could be endpoint). So it isn't crossing, because they don't go through each other. The lines can be at any angle as well, not just vertical and horizontal like in my drawings